peer/brontide.go

```diff
@@ -532,6 +532,11 @@ type Brontide struct {
 	// off new wire messages for handing.
 	msgRouter fn.Option[msgmux.Router]
 
+	// globalMsgRouter is a flag that indicates whether we have a global
+	// msg router. If so, then we don't worry about stopping the msg router
+	// when a peer disconnects.
+	globalMsgRouter bool
```

Comment:
this field is totally unnecessary. I think we just inline the call to `cfg.MsgRouter.IsSome()`

Reply:
Ah yeah can do that too: we can check against the config version as it has a life cycle independent of the peer itself.